common/compositor/va/varenderer.cpp

```diff
@@ -227,106 +228,127 @@ bool VARenderer::Draw(const MediaState& state, NativeSurface* surface) {
+    output_region.x = display_frame.left;
+    output_region.y = display_frame.top;
+    output_region.width = layer_in->GetDisplayFrameWidth();
+    output_region.height = layer_in->GetDisplayFrameHeight();
+
+
+    VABlendState bs = {};
+    bs.flags = VA_BLEND_PREMULTIPLIED_ALPHA;
+
+    VAProcPipelineParameterBuffer pipe_param = {};
+    pipe_param.surface = surface_in;
+    pipe_param.surface_region = &surface_region;
+    pipe_param.surface_color_standard = VAProcColorStandardBT601;
+    pipe_param.output_region = &output_region;
+    pipe_param.output_color_standard = VAProcColorStandardBT601;
+    pipe_param.blend_state = &bs;
+
+    DUMPTRACE("surface_region: (%d, %d, %d, %d)\n", surface_region.x,
+              surface_region.y, surface_region.width, surface_region.height);
+    DUMPTRACE("Layer DisplayFrame:(%d,%d,%d,%d)\n", output_region.x,
+              output_region.y, output_region.width, output_region.height);
 
 #if VA_MAJOR_VERSION >= 1
-  // currently rotation is only supported by VA on Android.
-  uint32_t rotation = 0, mirror = 0;
-  HWCTransformToVA(state.layer_->GetTransform(), rotation, mirror);
-  pipe_param.rotation_state = rotation;
-  pipe_param.mirror_state = mirror;
+    // currently rotation is only supported by VA on Android.
+    uint32_t rotation = 0, mirror = 0;
+    HWCTransformToVA(layer_in->GetTransform(), rotation, mirror);
+    pipe_param.rotation_state = rotation;
+    pipe_param.mirror_state = mirror;
 #endif
 
-  ScopedVABufferID pipeline_buffer(va_display_);
-  if (!pipeline_buffer.CreateBuffer(
+    for (auto itr = state.colors_.begin(); itr != state.colors_.end(); itr++) {
+      SetVAProcFilterColorValue(itr->first, itr->second);
+    }
+    SetVAProcFilterDeinterlaceMode(state.deinterlace_);
+
+    if (!UpdateCaps()) {
+      ETRACE("Failed to update capabailities. \n");
+      return false;
+    }
+
+    pipe_param.filter_flags = GetVAProcFilterScalingMode(state.scaling_mode_);
+    if (filters_.size()) {
+      pipe_param.filters = filters_.data();
+    }
+    pipe_param.num_filters = static_cast<unsigned int>(filters_.size());
+
+    if (!pipeline_buffer.CreateBuffer(
           va_context_, VAProcPipelineParameterBufferType,
           sizeof(VAProcPipelineParameterBuffer), 1, &pipe_param)) {
-    return false;
-  }
+      return false;
+    }
 
-  VAStatus ret = VA_STATUS_SUCCESS;
-  ret = vaBeginPicture(va_display_, va_context_, surface_out);
-  ret |=
+    ret |=
       vaRenderPicture(va_display_, va_context_, &pipeline_buffer.buffer(), 1);
-  ret |= vaEndPicture(va_display_, va_context_);
 
-  if (surface_region.width == 1920 && surface_region.height == 1080) {
-    // FIXME: WA for OAM-63127. Not sure why this is needed but seems
-    // to ensure we have consistent 60 fps.
-    vaSyncSurface(va_display_, surface_out);
   }
 
+  ret |= vaEndPicture(va_display_, va_context_);
+
+  // reviewer: How to resolve this?
```

Comment:
OK, We can drop here. 
But when Android KPI is not meet, we should pick this

Reply:
Sure, will drop it.
BTW, this seems like a hack. Any idea how to properly resolve it?